### ISSUE TITLE
Noise on pin 42 (o'scope ch. 2) due to signal on pin 40 (ch. 1)

### ISSUE THREAD

**Issue by dbsamithey** [EspoTek/Labrador#256]
I noticed that noise appears on ch. 2 of the oscilloscope when a signal with frequency >~ 1 Khz and peak-to-peak magnitude >~ 2 V is applied to ch. 1.  As shown in the gif below, it strangely goes away when the ch. 1 signal has a peak-to-peak magnitude  >~3.2 V. This threshold appears to be insensitive to the PSU setting.  Also, the noise appears irrespective of whether the DC or AC pins of the signal generator and oscilloscope are used.
The same type of noise doesn't seem to show up on ch. 1 under any conditions.  On the other hand, if the hardware pins assigned to ch. 1 and ch. 2 are swapped in the board firmware, the noise starts to appear on ch. 1 instead of ch. 2.  I'm surprised about this hardware pin specificity because the input stage on the schematic looks identical for the two pins.  Please let me know if you have an idea as to how the noise might be mitigated.
![labrador_noise](https://github.com/EspoTek/Labrador/assets/127083957/ad499494-ff71-4a32-b1e1-240b29497112)

**Comment by EspoTek** [comment]
This is bizarre.

I know there's a "Signal Gen Triple" setting that pumps the signal gen output through an amplifier when >3.2V, but if anything this should reduce the noise when <3.2V.  Do you see the same artifacting when on a period that's not an integer multiple/divisor of 1ms?  (e.g. 4691Hz)?

**Comment by dbsamithey** [comment]
I still see the problem at that frequency.  Also, the threshold for the onset of the noise does look to be exactly 3.2 V, so the "Signal Gen Triple" setting seems relevant.  But, if the noise were purely related to the signal generation, then I would maybe expect it to show up regardless of whether the signal generator is plugged into channel 1.  In reality, the noise goes away when the signal gen is disconnected even when the output setting is kept at some large voltage ~< 3.2 V.
This suggests that the noise could be related to the measurement of the large waveform.  I checked this by analyzing the smaller signal in the default 1 channel mode of the Labrador while the larger signal was applied to the unsampled oscilloscope input.  The noise only ever appears on the pin 42 input, while the default 1 channel mode shows data from the pin 40 input, so for this test I again flashed a modified firmware with the hardware pins assigned such that the pin 42 input was shown in the 1 channel mode.  Then, with pin 42 measuring a ~400 mV waveform and a ~3.1 V waveform applied to pin 40, again the noise appeared.  So, the presence of a large signal on pin 40 is sufficient for the noise to appear, and actual measurement of the waveform is irrelevant.
Maybe the noise goes away when the voltage setting of the large signal is > 3.2 V because then the chip puts out a signal that is three times smaller.  The noise does in fact appear to have the same level when the voltage setting of the signal applied to pin 40 is 3.21 V / 3 = 1.066 V as when it is 3.21 V.  As far as I understand, for both these settings, the peak-to-peak magnitude of the signal produced by the chip is 1.066 V, so this observation is consistent with the chip-generated voltage being relevant.  If that's true though, then it's all the more confusing that actual application of the signal to pin 40 is important.  The increment 3.19->3.21 V only substantially changes the conditions on the chip's signal gen output pin, so it seems like it should not be decisive whether the signal is applied to pin 40.

The screenshots below I think clarify things a bit.  One is with a wire connected between the ch. 1 (pin 40) input and some piece of metal that's picking up 60 hz mains hum and with nothing connected to the ch. 2 (pin 42) input.  This results in strong distortion of the ch. 2 input.  On the other hand, with ch. 2 instead picking up mains hum and with nothing connected to ch. 1, no strong distortion shows up on ch. 1.  It seems like, generally speaking, the ch. 2 input is more susceptible to distortion when there's a signal on ch. 1 than vice versa, and the max voltage amplitude on the chip's signal gen pins is just one variable that controls the level of the distortion.

Ch. 1 picking up mains hum; Ch. 2 not connected to anything
![lab_distort2](https://github.com/EspoTek/Labrador/assets/127083957/86f4f6e6-3468-4acd-9732-06eb78d43c52)

![lab_distort1](https://github.com/EspoTek/Labrador/assets/127083957/3f923b5e-1a17-4589-bfe2-767c77da892b)
Ch. 2 picking up mains hum; Ch. 1 not connected to anything

**Comment by mi-hol** [comment]
@dbsamithey is this still an open issue?

**Comment by brentfpage** [comment]
One distinction between the two channels I see on the PCB is that, after CH1 is attenuated by ~75/1000 by the input voltage divider (Scope_CH1_DC -> Scope_Buffer_Input_CH1), it passes underneath the non-attenuated CH2 trace (Scope_CH2_DC).  In contrast, the attenuated signal Scope_Buffer_Input_CH2 never passes underneath Scope_CH1_DC .  I have zero experience with this sort of thing, but it seems at least plausible that this explains why CH2 signals produce more noise on CH1 than vice-versa.

**Comment by EspoTek** [comment]
> One distinction between the two channels I see on the PCB is that, after CH1 is attenuated by ~75/1000 by the input voltage divider (Scope_CH1_DC -> Scope_Buffer_Input_CH1), it passes underneath the non-attenuated CH2 trace (Scope_CH2_DC). In contrast, the attenuated signal Scope_Buffer_Input_CH2 never passes underneath Scope_CH1_DC . I have zero experience with this sort of thing, but it seems at least plausible that this explains why CH2 signals produce more noise on CH1 than vice-versa.

I don't think this explains the level of crosstalk that OP saw (it was a hardware fault and not common to all units), but good catch either way!

**Comment by brentfpage** [comment]
Saw the behavior depicted in the gif in the first post (posted by me btw on a different gh account) on two of my labradors, so ended up getting a few boards fabbed with slightly edited traces to see if I could mitigate the noise.  On both new boards, the ch. 2 unattenuated and ch. 1 attenuated scope input traces are routed such that they don't cross.  As shown in the gif below, this revision does appear to substantially reduce the amount of noise!  The revised traces could very well violate some design principle that I don't know about, but so far I haven't encountered any new issues.  The device functionality is just slightly extended by the revision, and more than anything I'm happy to satisfy my curiosity about the noise.

![Image](https://github.com/user-attachments/assets/c21523b8-bcfe-40ff-bfa6-1b10a2bb9698)

Fwiw, I think the possible origins of the remaining noise can be narrowed down based on the behavior of the two revised PCBs.  On one of them, pin 40 of the atxmega gets routed to the outer (closer to the board edge) scope input header and pin 42 of the atxmega gets routed to the inner scope input header (same as on the genuine PCB).  On the other revised board, this routing is swapped.  However, both revised boards tend to show more noise on the pin 42 channel than on the pin 40 channel.  Based on the commonalities and differences of the revised board traces, this suggests that the remaining noise is caused by something on Scope_Buffer_Input_Ch(1/2) or downstream.  Now I'm feeling lucky that the revisions made an improvement, because it appears the thing that I focused on wasn't the full story!
